### PR TITLE
Fix PersistenceIdsPublisher hung on failure messages

### LINKED ISSUE #6365: Akka.Persistence.Query.Sql: `PersistenceIds` hangs after about 2 hours
**Version Information**
Version of Akka.NET? v1.4.48
Which Akka.NET Modules? Akka.Persistence.Query.Sql and Akka.Persistence.SqlServer

**Describe the bug**

When running the following query:

https://github.com/petabridge/akkadotnet-code-samples/blob/ed35dfd4ee862e68afe6abdd825a9f3029d382ed/src/clustering/sharding-sqlserver/SqlSharding.Host/Actors/ProductIndexActor.cs#L80-L89

```csharp
/*
 * Kicks off an Akka.Persistence.Query instance that will continuously
 */
var query = Context.System.ReadJournalFor<SqlReadJournal>(SqlReadJournal.Identifier);
query.PersistenceIds()
    .Where(c => c.StartsWith(ProductTotalsActor.TotalsEntityNameConstant))
    .Select(c =>
    {
        var splitPivot = c.IndexOf("-", StringComparison.Ordinal);
        return new ProductFound(c[(splitPivot + 1)..]);
    })
    .To(Sink.ActorRef<ProductFound>(Self, Done.Instance))
    .Run(Context.Materializer());
```

This query will cease to produce new Akka.Stream items after approximately 2 hours (don't have an exact measurement) even though new PersistenceIds are being committed to the event journal. Restarting the query will reproduce all of the missing events.

**To Reproduce**
1. Clone https://github.com/petabridge/akkadotnet-code-samples/tree/master/src/clustering/sharding-sqlserver
2. Follow the instructions to run the project in the `README.md`
3. Go to http://localhost:5001
4. Add some entries
5. Look at the index page
6. Wait 2 hours
7. Add additional entries
8. Look at index page again (new entries added in step 7 will not be present)

**Expected behavior**

Query should continue producing new events indefinitely.

**Actual behavior**

Query stopped, raised no errors, and stopped producing 

**Environment**

SQL Server latest running on Linux, .NET 6 running on Linux - although I don't think it matters.

### PR DESCRIPTION
**Pull request by Arkatufus**
Fixes #6365

## Changes

* Change `Tell` to `Ask`
* Handle failures to maintain consistent actor receive state

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
@Arkatufus looks like a compilation failure

**Comment by Arkatufus** [comment]
Ah, API Verify problem

**Comment by Arkatufus** [comment]
Fixed

File: src/contrib/persistence/Akka.Persistence.Query.Sql/AllPersistenceIdsPublisher.cs [hunk 1]
```diff
@@ -64,9 +71,26 @@ private bool Initializing(object message)
                     Become(Active);
                     Stash.UnstashAll();
                     return true;
+                
                 case Cancel _:
                     Context.Stop(Self);
                     return true;
+                
+                case Status.Failure msg:
```
Comment (Aaronontheweb) [inline src/contrib/persistence/Akka.Persistence.Query.Sql/AllPersistenceIdsPublisher.cs:79]: LGTM

File: src/contrib/persistence/Akka.Persistence.Query.Sql/AllPersistenceIdsPublisher.cs [hunk 2]
```diff
@@ -77,14 +101,20 @@ private bool Active(object message)
         {
             switch (message)
             {
+                case CurrentPersistenceIds _:
+                    // Ignore duplicate CurrentPersistenceIds response
```
Comment (Aaronontheweb) [inline src/contrib/persistence/Akka.Persistence.Query.Sql/AllPersistenceIdsPublisher.cs:105]: LGTM

File: src/contrib/persistence/Akka.Persistence.Query.Sql/SqlReadJournal.cs [hunk 3]
```diff
@@ -26,7 +25,7 @@ public class SqlReadJournal :
         IAllEventsQuery,
         ICurrentAllEventsQuery
     {
-        public static string Identifier = "akka.persistence.query.journal.sql";
+        public const string Identifier = "akka.persistence.query.journal.sql";
```
Comment (Aaronontheweb) [inline src/contrib/persistence/Akka.Persistence.Query.Sql/SqlReadJournal.cs:28]: LGTM

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Comment by Aaronontheweb** [comment]
@Arkatufus this will need backport to v1.4

**Comment by Arkatufus** [comment]
Will do

**Pull request merged**